### PR TITLE
update cats & downstream dependencies

### PR DESCRIPTION
**Pull request by pepegar**
- Update cats-effect and downstream dependencies.
- Don't support 2.11 anymore.

### CODE REVIEW

File: hammock-asynchttpclient/src/test/scala/hammock/asynchttpclient/AsyncHttpClientInterpreterSpec.scala [hunk 1]
```diff
@@ -3,15 +3,16 @@ package asynchttpclient
...
 import org.asynchttpclient._
 import io.netty.handler.codec.http.{DefaultHttpHeaders, HttpHeaders}
 import io.netty.handler.codec.http.cookie.Cookie
-import org.scalatest.mockito._
+import org.scalatestplus.mockito._
 import AsyncHttpClientInterpreter._
 import scala.collection.JavaConverters._
 
-class AsyncHttpClientInterpreterSpec extends WordSpec with Matchers with MockitoSugar {
+class AsyncHttpClientInterpreterSpec extends AnyWordSpec with Matchers with MockitoSugar {
```
Comment (AlvaroCaste) [inline hammock-asynchttpclient/src/test/scala/hammock/asynchttpclient/AsyncHttpClientInterpreterSpec.scala]: Out of curiosity: do you use `MockitoSugar`?

Comment (pepegar) [reply hammock-asynchttpclient/src/test/scala/hammock/asynchttpclient/AsyncHttpClientInterpreterSpec.scala]: Yeah, I use mockito to mock the integration's http driver... I'd love to find a better approach though :)

Comment (AlvaroCaste) [reply hammock-asynchttpclient/src/test/scala/hammock/asynchttpclient/AsyncHttpClientInterpreterSpec.scala]: I am not a big fan of mocking tools but I've heard about http://scalamock.org/
we can take a look after this PR if you want :)

Comment (pepegar) [reply hammock-asynchttpclient/src/test/scala/hammock/asynchttpclient/AsyncHttpClientInterpreterSpec.scala]: Oh thanks! Yes please, go ahead :)

**Comment by pepegar** [comment]
The problem we're having now is that it doesn't build for `2.11` because monocle doesn't work for 2.11 as per 2.0.0... @llfrometa89 @miciek is anybody using this on `2.11`? do you have any issue if we don't support it anymore?

**Comment by AlvaroCaste** [comment]
> The problem we're having now is that it doesn't build for `2.11` because monocle doesn't work for 2.11 as per 2.0.0... @llfrometa89 @miciek is anybody using this on `2.11`? do you have any issue if we don't support it anymore?

you should then update this
https://github.com/pepegar/hammock/blob/master/README.md#L18

**Pull request merged**